src/core/src/core_logic/ExecutionConfig.py

```diff
@@ -44,6 +44,9 @@ def __init__(self, env_layer, composite_logger, execution_parameters):
         self.config_folder = self.environment_settings[Constants.EnvSettings.CONFIG_FOLDER]
         self.status_folder = self.environment_settings[Constants.EnvSettings.STATUS_FOLDER]
         self.events_folder = self.environment_settings[Constants.EnvSettings.EVENTS_FOLDER]
+        self.temp_folder = self.environment_settings[Constants.EnvSettings.TEMP_FOLDER]
+        # todo: Q or point of discussion for reviewer: should we add a none and/or path exists check on temp folder? If yes, what is core supposed to do in that scenario?
```

Comment:
Handler is responsible for ensuring the folder exists. Core should never remove the folder itself. Only contents it's written.

Reply:
If core sees the folder being non-existent, it may create it once and then log that. This is more likely to happen for auto-assessment (intuitively) due to timing issues and should never happen for mainline Core execution if we've gotten things right.